Mathlib/CategoryTheory/Comma/Over/Pullback.lean

```diff
@@ -82,9 +83,28 @@ def pullbackComp {X Y Z : C} (f : X ⟶ Y) (g : Y ⟶ Z) :
   conjugateIsoEquiv (mapPullbackAdj _) ((mapPullbackAdj _).comp (mapPullbackAdj _))
     (Over.mapComp _ _).symm
 
-instance pullbackIsRightAdjoint {X Y : C} (f : X ⟶ Y) : (pullback f).IsRightAdjoint  :=
+instance pullbackIsRightAdjoint {X Y : C} (f : X ⟶ Y) : (pullback f).IsRightAdjoint :=
   ⟨_, ⟨mapPullbackAdj f⟩⟩
 
+open pullback in
+/-- A functor `F : T ⥤ D` induces a functor `Over X ⥤ Over (F.obj X)` in the obvious way. -/
+@[simps]
+def postAdjunctionLeft {F : C ⥤ D} {G : D ⥤ C} (a : F ⊣ G) :
+    post F ⊣ post G ⋙ pullback (a.unit.app X) where
+  unit.app A := homMk <| lift (a.unit.app A.left) A.hom (by aesop_cat)
+  counit.app A := homMk (F.map (fst _ _) ≫ a.counit.app A.left) <| by
+    simp only [Functor.comp_obj, post_obj, Functor.id_obj, pullback_obj_left, mk_left, mk_hom,
+      pullback_obj_hom, Functor.const_obj_obj, assoc]
+    calc
+          F.map (fst (G.map A.hom) (a.unit.app X)) ≫ a.counit.app A.left ≫ A.hom
+      _ = F.map (fst (G.map A.hom) (a.unit.app X) ≫ G.map A.hom) ≫ a.counit.app (F.obj X) := by
+        simp
+      _ = F.map (snd (G.map A.hom) (a.unit.app X) ≫ a.unit.app X) ≫ a.counit.app (F.obj X) := by
+        rw [condition]
+      _ = F.map (snd (G.map A.hom) (a.unit.app X)) := by simp
+  counit.naturality {A B} f := by ext; simp [← Functor.map_comp_assoc, -Functor.map_comp]; simp
+  left_triangle_components A := by ext; simp [← Functor.map_comp_assoc, -Functor.map_comp]
```

Comment:
```suggestion
    post F ⊣ post G ⋙ pullback (a.unit.app X) :=
  ((mapPullbackAdj (a.unit.app X)).comp (postAdjunctionRight (Y := F.obj X) a)).ofNatIsoLeft
    (NatIso.ofComponents (fun Y ↦ Over.isoMk (.refl _)))
```

Reply:
And can you also add `post.instIsRightAdjoint` like you did to `postAdjunctionLeft`? Also I'm not sure if it should be camel case.